src/rmc-compiler/src/main.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This is the main entry point to our compiler driver. This code accepts a few options that
+//! can be used to configure goto-c compilation as well as all other flags supported by rustc.
+//!
+//! Like miri, clippy, and other tools developed on the top of rustc, we rely on the
+//! rustc_private feature and a specific version of rustc.
+
+#![feature(rustc_private, once_cell)]
+extern crate rustc_codegen_ssa;
+extern crate rustc_driver;
+extern crate rustc_session;
+
+use clap::{
+    app_from_crate, crate_authors, crate_description, crate_name, crate_version, App, AppSettings,
+    Arg, ArgMatches,
+};
+use rmc_queries::{QueryDb, UserInput};
+use rustc_driver::{init_env_logger, install_ice_hook, Callbacks, RunCompiler};
+use std::path::PathBuf;
+use std::rc::Rc;
+
+/// This function generates all rustc configurations required by our goto-c codegen.
+fn rustc_gotoc_flags(lib_path: &str) -> Vec<String> {
+    let rmc_deps = lib_path.clone().to_owned() + "/deps";
+    let args = vec![
+        "-C",
+        "overflow-checks=on",
+        "-C",
+        "panic=abort",
+        "-Z",
+        "panic_abort_tests=yes",
+        "-Z",
+        "trim-diagnostic-paths=no",
+        "-Z",
+        "human_readable_cgu_names",
+        "--cfg=rmc",
+        "-Z",
+        "crate-attr=feature(register_tool)",
+        "-Z",
+        "crate-attr=register_tool(rmctool)",
+        "-L",
+        lib_path,
+        "--extern",
+        "rmc",
+        "-L",
+        rmc_deps.as_str(),
+    ];
+    args.iter().map(|s| s.to_string()).collect()
+}
+
+fn parser<'a, 'b>() -> App<'a, 'b> {
+    app_from_crate!()
```

Comment:
Do you have a preference for `clap`? I was starting work on cargo-rmc with `structopt` which is a wrapper around `clap`. Just curious if you think I shouldn't be using that for some reason (it's not too late for me to abort and turn around...)

Reply:
I don't have any strong preference. I haven't done much research since this code is throw away code. Whatever works.